### PR TITLE
UNITY-492 Hand Binder Editor Variables

### PR DESCRIPTION
**Pull request by MattGrayUL**
## Pull Request Templates

Switch template by going to preview and clicking the link - note it not work if you've made any changes to the description.

- [default.md](?expand=1) - for contributions to stable packages.
- [lightweight.md](?expand=1&template=lightweight.md) - for contributions to pre-release/preview packages use the lightweight merge request template. The quality threshold for reviewing is lower - it prioritizes a lean review process.
- [release.md](?expand=1&template=release.md) - for release merge requests.

**You are currently using: default.md**

Note: these links work by overwriting query parameters of the current url. If the current url contains any you may want to amend the url with `&template=name.md` instead of using the link. See [query parameter docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request) for more information.

## Summary

This PR removes the public access to variables that are used in the editor script only.

## Contributor Tasks

_These tasks are for the merge request creator to tick off when creating a merge request._

- [x] Pair review with a member of the QA team.
- [x] Add any release testing considerations to the MR for the next release.
- [x] Check any relevant CHANGELOG files have been updated.
- [x] Ensure documentation requirements are met e.g., public API is commented.
- [x] Consider any licensing/other legal implications for this MR e.g. notices required by any new libraries.
- [x] Add any relevant labels such as `breaking` to this MR.
- [ ] If this MR closes a Jira issue, make sure the fix version on the JIRA issue is set to the correct one.

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

[Use emojis in review threads to communicate intent and help contributors.](https://github.com/ultraleap/UnityPlugin/blob/develop/CONTRIBUTING.md#review-threads)

- [x] Code reviewed.
- [x] Non-code assets e.g. Unity assets/scenes reviewed.
- [x] Documentation has been reviewed. Includes checking documentation requirements are met and not missing e.g., public API is commented.
- [ ] Checked and agree with release testing considerations added to MR for the next release.

## Closes JIRA Issue

_If this MR closes any JIRA issues list them below in the form `Closes PROJECT-#`_

### CODE REVIEW

File: Packages/Tracking/Hands/Runtime/Scripts/HandBinder/HandBinder.cs [hunk 1]
```diff
@@ -526,50 +526,46 @@ bool CanUseScaleFeature()
...
+        [SerializeField] bool FineTuning;
         /// <summary>  
         /// Used by the editor script. The DebugOptions allow to show a debug hand in the scene view
         /// and visualize its rotation and its attached gameobjects
         /// </summary>
-        public bool DebugOptions;
-        /// <summary> 
-        /// Used by the editor script. 
-        /// </summary>
-        public bool EditPoseNeedsResetting = false;
```
Comment (JuliaRonnebergerUL) [inline Packages/Tracking/Hands/Runtime/Scripts/HandBinder/HandBinder.cs:572]: Can this be added back in? It is used in the handBinder

Comment (MattGrayUL) [reply Packages/Tracking/Hands/Runtime/Scripts/HandBinder/HandBinder.cs:572]: oops, not sure how I managed that! it sure can

Comment (MattGrayUL) [reply Packages/Tracking/Hands/Runtime/Scripts/HandBinder/HandBinder.cs:572]: Actually, I meant to _also_ remove the usage as it's not _actually_ used, it is just set and then unset without being used in a comparison anywhere

**Comment by JuliaRonnebergerUL** [comment]
Can we somehow get rid of these warnings? Seems like unity doesn't know that it is being used in the editor script..

Screenshot 2022-07-22 092337

**Review by JuliaRonnebergerUL — APPROVED** [review]

**Comment by MattGrayUL** [comment]
@Greg-UL This PR makes access to some editor-only variables private, i don't believe there are any testing considerations to add

Just to be double safe, when making the PR I ran the Binder on our generic hands to ensure it still acts as expected and it did

Are you happy with that? :)

**Comment by MattGrayUL** [comment]
> I think so. Is it likely this change could have impacted on the autoscaling functionality at all?

it won't have any impact on the autoscaling :)

**Comment by Greg-UL** [comment]
> > I think so. Is it likely this change could have impacted on the autoscaling functionality at all?
> 
> it won't have any impact on the autoscaling :)

then I'm happy - please proceed

**Pull request merged**